pkg/streamingpromql/planning.go

```diff
@@ -29,6 +32,14 @@ import (
 	"github.com/grafana/mimir/pkg/util/spanlogger"
 )
 
+var smoothedIncompatibleFunctionPrefix string
+var anchoredIncompatibleFunctionPrefix string
+
+func init() {
+	smoothedIncompatibleFunctionPrefix = fmt.Sprintf("smoothed modifier can only be used with: %s - not with ", sortImplode(promql.SmoothedSafeFunctions))
+	anchoredIncompatibleFunctionPrefix = fmt.Sprintf("anchored modifier can only be used with: %s - not with ", sortImplode(promql.AnchoredSafeFunctions))
```

Comment:
WDYT about making a base `ErrAnchoredIncompatibleFunction` and wrapping with the specific function below?